data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/admin/Readiness.java

```diff
@@ -60,10 +79,27 @@ public Readiness(final DataProvider provider) {
   @Override
   public void handle(final Context ctx) throws Exception {
     ctx.header(Header.CACHE_CONTROL, CACHE_NONE);
-    if (!chainDataProvider.isStoreAvailable() || syncProvider.isSyncing()) {
+    if (!chainDataProvider.isStoreAvailable()
+        || syncProvider.isSyncing()
+        || !reachedTargetPeerCount(ctx)) {
       ctx.status(SC_SERVICE_UNAVAILABLE);
     } else {
       ctx.status(SC_OK);
     }
   }
+
+  private boolean reachedTargetPeerCount(final Context ctx) {
+    try {
+      final Map<String, List<String>> parameters = ctx.queryParamMap();
+      if (parameters.isEmpty()) return true; // Backwards compatibility
+      LOG.debug("Current peer count is {}", networkDataProvider.getPeerCount());
+      return !parameters.isEmpty()
+          && parameters.containsKey(TARGET_PEER_COUNT)
```

Comment:
nit: Can remove the `!parameters.isEmpty()` check here as `parameters.containsKey` will always return false for an empty map.

Reply:
Good point, I thought of that. 
I'll update the code.